code/modules/planet/weather.dm

```diff
@@ -0,0 +1,268 @@
+		WEATHER_BLIZZARD = 20,
+		WEATHER_HAIL = 5,
+		WEATHER_OVERCAST = 5
+		)
+
+/datum/weather/sif/snow/process_effects()
+	for(var/turf/simulated/floor/outdoors/snow/S in outdoor_turfs)
+		for(var/dir_checked in cardinal)
+			var/turf/simulated/floor/T = get_step(S, dir_checked)
+			if(istype(T))
+				if(istype(T, /turf/simulated/floor/outdoors) && prob(33))
+					T.chill()
+
+/datum/weather/sif/blizzard
+	name = "blizzard"
+	icon_state = "snowfall_heavy"
+	temp_high = 223.15 // -40c
+	temp_low = 203.15  // -60c
+	light_modifier = 0.3
+	transition_chances = list(
+		WEATHER_SNOW = 45,
+		WEATHER_BLIZZARD = 40,
+		WEATHER_HAIL = 10,
+		WEATHER_OVERCAST = 5
+		)
+
+/datum/weather/sif/blizzard/process_effects()
+	for(var/turf/simulated/floor/outdoors/snow/S in outdoor_turfs)
+		for(var/dir_checked in cardinal)
+			var/turf/simulated/floor/T = get_step(S, dir_checked)
+			if(istype(T))
+				if(istype(T, /turf/simulated/floor/outdoors) && prob(50))
+					T.chill()
+
+/datum/weather/sif/rain
+	name = "rain"
+	icon_state = "rain"
+	light_modifier = 0.5
+	transition_chances = list(
+		WEATHER_OVERCAST = 25,
+		WEATHER_LIGHT_SNOW = 10,
+		WEATHER_RAIN = 50,
+		WEATHER_STORM = 10,
+		WEATHER_HAIL = 5
+		)
+
+/datum/weather/sif/rain/process_effects()
+	for(var/mob/living/L in living_mob_list)
+		if(L.z in holder.our_planet.expected_z_levels)
+			var/turf/T = get_turf(L)
+			if(!T.outdoors)
+				return // They're indoors, so no need to rain on them.
+
+			L.adjust_fire_stacks(-5)
+			to_chat(L, "<span class='warning'>Rain falls on you.</span>")
+
+/datum/weather/sif/storm
+	name = "storm"
+	icon_state = "storm"
+	temp_high = 233.15 // -30c
+	temp_low = 213.15  // -50c
+	light_modifier = 0.3
+	transition_chances = list(
+		WEATHER_RAIN = 45,
+		WEATHER_STORM = 40,
+		WEATHER_HAIL = 10,
+		WEATHER_OVERCAST = 5
+		)
+
+/datum/weather/sif/rain/process_effects()
+	for(var/mob/living/L in living_mob_list)
+		if(L.z in holder.our_planet.expected_z_levels)
+			var/turf/T = get_turf(L)
+			if(!T.outdoors)
+				return // They're indoors, so no need to rain on them.
+
+			L.adjust_fire_stacks(-10)
+			to_chat(L, "<span class='warning'>Rain falls on you, drenching you in water.</span>")
+
+/datum/weather/sif/hail
+	name = "hail"
+	icon_state = "hail"
+	temp_high = 233.15 // -30c
+	temp_low = 213.15  // -50c
+	light_modifier = 0.3
+	transition_chances = list(
+		WEATHER_RAIN = 45,
+		WEATHER_STORM = 10,
+		WEATHER_HAIL = 40,
+		WEATHER_OVERCAST = 5
+		)
+
+/datum/weather/sif/hail/process_effects()
+	for(var/mob/living/L in living_mob_list)
+		if(L.z in holder.our_planet.expected_z_levels)
+			var/turf/T = get_turf(L)
+			if(!T.outdoors)
+				return // They're indoors, so no need to pelt them with ice.
+
+			var/target_zone = pick(BP_ALL)
+			var/amount_blocked = L.run_armor_check(target_zone, "melee")
+
+			if(amount_blocked >= 100)
+				return // No need to apply damage.
+
+			L.apply_damage(rand(5, 10), BRUTE, target_zone, amount_blocked, used_weapon = "hail")
```

Comment:
Autohealing will probably remove the damage regardless but I can tweak it.  Also remember it only fires once a minute.